test/app/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+// simple test application
+// allows to check if users can use the provided connectionstring,
+// and if data remains after changes are made
+
+var (
+	collection *mongo.Collection
+	ctx        = context.TODO()
+
+	// related to DB
+	dbName         = "Ships"
+	collectionName = "ShipSpec"
+	port           = 8080
+)
+
+func main() {
+	r := newRouter()
+	http.ListenAndServe(fmt.Sprintf(":%d", port), r)
+}
+
+func newRouter() *mux.Router {
+	r := mux.NewRouter()
+	r.HandleFunc("/", handler).Methods("GET")
+	r.HandleFunc("/mongo/", getKeyValue).Methods("GET")
+	r.HandleFunc("/mongo/", postKeyValue).Methods("POST")
+	r.HandleFunc("/mongo/", deleteKeyValue).Methods("DELETE")
+	return r
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "It is working")
+}
+
+func getKeyValue(w http.ResponseWriter, r *http.Request) {
+	ship := getShipFromRequest(r)
```

Comment:
what do you think if we pass the `id` parameter instead of passing the whole `Ship` object? It seems that we only need to find the object in database by id?

Reply:
I was thinking maybe we could add UI for manual test/demonstration purposes later. Or we don't need that ?